parts/k8s/cloud-init/artifacts/cse_install.sh

```diff
@@ -31,6 +31,7 @@ installDeps() {
     retrycmd_if_failure_no_stats 120 5 25 curl -fsSL https://packages.microsoft.com/config/ubuntu/${UBUNTU_RELEASE}/packages-microsoft-prod.deb > /tmp/packages-microsoft-prod.deb || exit $ERR_MS_PROD_DEB_DOWNLOAD_TIMEOUT
     retrycmd_if_failure 60 5 10 dpkg -i /tmp/packages-microsoft-prod.deb || exit $ERR_MS_PROD_DEB_PKG_ADD_FAIL
     apt_get_update || exit $ERR_APT_UPDATE_TIMEOUT
+    apt_get_dist_upgrade || exit $ERR_APT_DIST_UPGRADE_TIMEOUT
```

Comment:
Agreed we don't want to run this in the VHD case systematically since that would break the no outbound scenario. However, `apt-get dist-upgrade` seems to be most relevant when building from a VHD that was built a few days ago as the OS image might be outdated by then. For clusters deployed using the "full install" or ubuntu distro flows, `dist-upgrade` isn't as relevant since the VM will have just been built using the latest image available on Marketplace.

Reply:
"latest image available" is always a moving target, though, so this fills a non-trivial gap during VHD build time: if, say, the image version we're building upon is n days old this will add an additional n days of apt hygiene to the dependency graph.

Reply:
Makes sense. Also I hadn't realized this is actually running as part of the VHD build too since it's in installDeps. The PR title is misleading, it implies it's only for the full install flow.